### PR TITLE
libratbag: fix ratbagctl execution

### PR DESCRIPTION
**Pull request by Atemu**
It requires gobject-introspection, don't ask me why a CLI has any need for that.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 1]
```diff
@@ -12,14 +12,14 @@ stdenv.mkDerivation rec {
     rev    = "v${version}";
     sha256 = "sha256-TQ8DVj4yqq3IA0oGnLDz+QNTyNRmGqspEjkPeBmXNew=";
   };
-
   nativeBuildInputs = [
-    meson ninja pkg-config gitMinimal swig check valgrind
+    meson ninja pkg-config gitMinimal swig check valgrind wrapGAppsHook
```
Comment (eclairevoyant) [inline pkgs/os-specific/linux/libratbag/default.nix:16]:
```suggestion
    meson ninja pkg-config gitMinimal swig check valgrind wrapGAppsHook3
```

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 2]
```diff
@@ -12,14 +12,14 @@ stdenv.mkDerivation rec {
     rev    = "v${version}";
     sha256 = "sha256-TQ8DVj4yqq3IA0oGnLDz+QNTyNRmGqspEjkPeBmXNew=";
   };
-
```
Comment (eclairevoyant) [inline pkgs/os-specific/linux/libratbag/default.nix:15]: nit: I wouldn't remove this line, for readability

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/os-specific/linux/libratbag/default.nix (hunk 2 above)
Comment (Atemu) [reply pkgs/os-specific/linux/libratbag/default.nix:15]: Not sure where it went honestly

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 3]
```diff
@@ -14,12 +14,13 @@ stdenv.mkDerivation rec {
...
 
   nativeBuildInputs = [
-    meson ninja pkg-config gitMinimal swig check valgrind
+    meson ninja pkg-config gitMinimal swig check valgrind wrapGAppsHook3
   ];
 
   buildInputs = [
     glib systemd udev libevdev json-glib libunistring
     (python3.withPackages (ps: with ps; [ evdev pygobject3 ]))
+    gobject-introspection # ratbagctl needs this
```
Comment (jtojnar) [inline pkgs/os-specific/linux/libratbag/default.nix:23]: It needs it for setup hook, so it should go to `nativeBuildInputs`.

File: pkgs/os-specific/linux/libratbag/default.nix [hunk 4]
```diff
@@ -14,12 +14,13 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [
-    meson ninja pkg-config gitMinimal swig check valgrind
+    meson ninja pkg-config gitMinimal swig check valgrind wrapGAppsHook3
```
Comment (jtojnar) [inline pkgs/os-specific/linux/libratbag/default.nix:17]: For the wrapper to find `glib` typelibs, `wrapGAppsNoGuiHook` should be sufficient

**Review by jtojnar — CHANGES_REQUESTED** [review]
This is fallout from `gobject-introspection` moving `glib` typelibs to glib package, see https://github.com/NixOS/nixpkgs/pull/306080#issuecomment-2081520383

**Comment by eclairevoyant** [comment]
fixed in #316717

**Pull request closed without merging**